FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Scheduler/Jobs/JobManager.cs

```diff
@@ -33,20 +34,23 @@ public class JobManager : IDisposable, IAsyncDisposable
             IJobStore jobStore,
             ITaskExecutor taskExecutor,
             IFhirSpecificationProvider fhirSpecificationProvider,
+            JobProgressUpdaterFactory jobProgressUpdaterFactory,
             IOptions<JobSchedulerConfiguration> schedulerConfiguration,
             IOptions<JobConfiguration> jobConfiguration,
             ILogger<JobManager> logger)
         {
             EnsureArg.IsNotNull(jobStore, nameof(jobStore));
             EnsureArg.IsNotNull(taskExecutor, nameof(taskExecutor));
             EnsureArg.IsNotNull(fhirSpecificationProvider, nameof(fhirSpecificationProvider));
+            EnsureArg.IsNotNull(jobProgressUpdaterFactory, nameof(jobProgressUpdaterFactory));
             EnsureArg.IsNotNull(schedulerConfiguration, nameof(schedulerConfiguration));
             EnsureArg.IsNotNull(jobConfiguration, nameof(jobConfiguration));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
             _jobStore = jobStore;
             _taskExecutor = taskExecutor;
             _fhirSpecificationProvider = fhirSpecificationProvider;
+            _jobProgressUpdaterFactory = jobProgressUpdaterFactory;
```

Comment:
Seems here already exist a private **_jobStore**, can we directly use it to create **jobProgressUpdater**?

Reply:
Moved the factory to `jobExecutor` class and we need the factory to resolve dependencies.